File: app/src/main/java/com/chumian/browser/settings/SettingsManager.kt
package com.chumian.browser.settings

import android.content.Context
import android.content.SharedPreferences

class SettingsManager(context: Context) {
    private val prefs: SharedPreferences = context.getSharedPreferences("settings", Context.MODE_PRIVATE)

    // 搜索引擎设置
    var searchEngine: String
        get() = prefs.getString("search_engine", "baidu") ?: "baidu"
        set(value) = prefs.edit().putString("search_engine", value).apply()

    // 主题设置
    var themeMode: String
        get() = prefs.getString("theme_mode", "system") ?: "system"
        set(value) = prefs.edit().putString("theme_mode", value).apply()

    // 字体大小设置
    var fontSize: Int
        get() = prefs.getInt("font_size", 100)
        set(value) = prefs.edit().putInt("font_size", value).apply()

    // 获取搜索URL
    fun getSearchUrl(query: String): String {
        return when (searchEngine) {
            "baidu" -> "https://www.baidu.com/s?wd=$query"
            "google" -> "https://www.google.com/search?q=$query"
            "bing" -> "https://www.bing.com/search?q=$query"
            "sogou" -> "https://www.sogou.com/web?query=$query"
            "360" -> "https://www.so.com/s?q=$query"
            else -> "https://www.baidu.com/s?wd=$query"
        }
    }

    // 获取搜索引擎名称
    fun getSearchEngineName(): String {
        return when (searchEngine) {
            "baidu" -> "百度"
            "google" -> "Google"
            "bing" -> "必应"
            "sogou" -> "搜狗"
            "360" -> "360搜索"
            else -> "百度"
        }
    }

    // 获取主题名称
    fun getThemeName(): String {
        return when (themeMode) {
            "light" -> "浅色"
            "dark" -> "深色"
            "system" -> "跟随系统"
            else -> "跟随系统"
        }
    }

    // 获取字体大小名称
    fun getFontSizeName(): String {
        return when (fontSize) {
            75 -> "较小"
            100 -> "默认"
            125 -> "较大"
            150 -> "大"
            else -> "默认"
        }
    }
}

File: ios/ChumianBrowser/Managers/SettingsManager.swift
import Foundation
import SwiftUI

enum SearchEngine: String, CaseIterable, Identifiable {
    case baidu = "baidu"
    case google = "google"
    case bing = "bing"
    case sogou = "sogou"
    case search360 = "360"
    
    var id: String { rawValue }
    
    var displayName: String {
        switch self {
        case .baidu: return "百度"
        case .google: return "谷歌"
        case .bing: return "必应"
        case .sogou: return "搜狗"
        case .search360: return "360搜索"
        }
    }
    
    var searchUrl: String {
        switch self {
        case .baidu: return "https://www.baidu.com/s?wd="
        case .google: return "https://www.google.com/search?q="
        case .bing: return "https://www.bing.com/search?q="
        case .sogou: return "https://www.sogou.com/web?query="
        case .search360: return "https://www.so.com/s?q="
        }
    }
    
    var homeUrl: String {
        switch self {
        case .baidu: return "https://www.baidu.com"
        case .google: return "https://www.google.com"
        case .bing: return "https://www.bing.com"
        case .sogou: return "https://www.sogou.com"
        case .search360: return "https://www.so.com"
        }
    }
}

enum ThemeMode: String, CaseIterable, Identifiable {
    case system = "system"
    case light = "light"
    case dark = "dark"
    
    var id: String { rawValue }
    
    var displayName: String {
        switch self {
        case .system: return "跟随系统"
        case .light: return "浅色"
        case .dark: return "深色"
        }
    }
    
    var colorScheme: ColorScheme? {
        switch self {
        case .system: return nil
        case .light: return .light
        case .dark: return .dark
        }
    }
}

class SettingsManager: ObservableObject {
    @Published var searchEngine: SearchEngine {
        didSet { UserDefaults.standard.set(searchEngine.rawValue, forKey: "searchEngine") }
    }
    
    @Published var themeMode: ThemeMode {
        didSet { UserDefaults.standard.set(themeMode.rawValue, forKey: "themeMode") }
    }
    
    @Published var fontSize: Int {
        didSet { UserDefaults.standard.set(fontSize, forKey: "fontSize") }
    }
    
    @Published var isDesktopMode: Bool {
        didSet { UserDefaults.standard.set(isDesktopMode, forKey: "isDesktopMode") }
    }
    
    @Published var homePage: String {
        didSet { UserDefaults.standard.set(homePage, forKey: "homePage") }
    }
    
    @Published var blockPopups: Bool {
        didSet { UserDefaults.standard.set(blockPopups, forKey: "blockPopups") }
    }
    
    @Published var saveHistory: Bool {
        didSet { UserDefaults.standard.set(saveHistory, forKey: "saveHistory") }
    }
    
    init() {
        let savedEngine = UserDefaults.standard.string(forKey: "searchEngine") ?? "baidu"
        self.searchEngine = SearchEngine(rawValue: savedEngine) ?? .baidu
        
        let savedTheme = UserDefaults.standard.string(forKey: "themeMode") ?? "system"
        self.themeMode = ThemeMode(rawValue: savedTheme) ?? .system
        
        let savedFontSize = UserDefaults.standard.integer(forKey: "fontSize")
        self.fontSize = savedFontSize > 0 ? savedFontSize : 100
        
        self.isDesktopMode = UserDefaults.standard.bool(forKey: "isDesktopMode")
        
        self.homePage = UserDefaults.standard.string(forKey: "homePage") ?? "https://www.baidu.com"
        
        self.blockPopups = UserDefaults.standard.object(forKey: "blockPopups") as? Bool ?? true
        
        self.saveHistory = UserDefaults.standard.object(forKey: "saveHistory") as? Bool ?? true
    }
    
    func getSearchUrl(_ query: String) -> String {
        let encodedQuery = query.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? query
        return searchEngine.searchUrl + encodedQuery
    }
    
    var desktopUserAgent: String {
        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Safari/605.1.15"
    }
    
    var mobileUserAgent: String {
        "Mozilla/5.0 (iPhone; CPU iPhone OS 17_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.0 Mobile/15E148 Safari/604.1"
    }
    
    func currentUserAgent(isDesktop: Bool) -> String {
        isDesktop ? desktopUserAgent : mobileUserAgent
    }
}
